src/main/java/com/ocelotslovebirds/birdhaus/blocks/BirdhouseBlockEntity.java

```diff
@@ -69,21 +75,49 @@ public void load(CompoundTag tag) {
      */
 
     public void tickServer() {
-        if (counter > 0) {
-            counter--;
-            setChanged();
+        handleSeedsForTick();
+        handleBirdSpawnForTick();
+    }
+
+    /*
+     * @param xOffset  x offset (East-West axis) at which to spawn the bird
+     * @param yOffset  y offset (Up-Down axis) at which to spawn the bird
+     * @param yOffset  z offset (North-South axis) at which to spawn the bird
+     * All offsets are relative to the birdhouse
+     * TODO: Check if yOffset actually works, it is possible that
+     * birds can only be spawned on the ground.
+    */
+    private void spawnNewBird(int xOffset, int yOffset, int zOffset) {
+        // Dirty type casting because it works
+        ServerLevel lvl = (ServerLevel) this.getLevel();
+        Parrot newBird = new Parrot(EntityType.PARROT, lvl);
+        lvl.addFreshEntity(newBird);
+        BlockPos desBlockPos = this.getBlockPos().offset(xOffset, yOffset, zOffset);
+
+        // Move the spawned bird to the destination
+        // 0.0 because it works, not sure what this supposed to be
+        newBird.moveTo(desBlockPos, (float) 0.0, (float) 0.0);
+    }
+
+    private void handleBirdSpawnForTick() {
+        if (tickerForBirdSpawns.tick()) {
+            int xOffset = ThreadLocalRandom.current().nextInt(-20, 20);
+            int yOffset = 50;
+            int zOffset = ThreadLocalRandom.current().nextInt(-20, 20);
+            spawnNewBird(xOffset, yOffset, zOffset);
         }
-        if (counter <= 0) {
-            if (!itemHandler.extractItem(0, 1, false).isEmpty()) {
-                counter = 100;
-                setChanged();
-            }
+    }
+
+    private void handleSeedsForTick() {
+        Boolean seedExtracted = false;
+        if (tickerForSeedConsumption.tick()) {
+            seedExtracted = !itemHandler.extractItem(0, 1, false).isEmpty();
         }
 
         // When adding seeds, if the seeds are "burnable" then set the block to active.
         BlockState blockState = getBlockState();
-        if (blockState.getValue(BlockStateProperties.CONDITIONAL) != counter > 0) {
-            level.setBlock(worldPosition, blockState.setValue(BlockStateProperties.CONDITIONAL, counter > 0),
+        if (blockState.getValue(BlockStateProperties.CONDITIONAL) != seedExtracted) {
+            level.setBlock(worldPosition, blockState.setValue(BlockStateProperties.CONDITIONAL, seedExtracted),
                     Block.UPDATE_ALL);
         }
```

Comment:
@Zachary-Concordia can you verify that I havent altered the behaviour of the birdhouse as far as seed consumption is concerned?

Reply:
Ok I think I screwed it up, okay so correct me if I'm wrong, here's my new interpretation of what your code was doing

1. Every 100 ticks, the block becomes ready to consume a seed (here's when a block becomes active)
2. Once it does consume a seed, it becomes inactive until another 100 ticks pass

Reply:
@Zachary-Concordia alright cool! I just changed the code to match the intended behaviour. I believe that code describes the intended behaviour more clearly now.

@shafiqimtiaz If you have a moment, could you retest this :) I think we got back the original behaviour